poulpy-hal/README.md

```diff
@@ -0,0 +1,87 @@
+# 🐙 Poulpy-HAL
+
+**Poulpy-HAL** is a Rust crate that provide backend agnostic layouts and trait-based low-level lattice arithmetic matching the API of [**spqlios-arithmetic**](https://github.com/tfhe/spqlios-arithmetic). This allows developpers to implement lattice-based schemes generically, with the ability to plug in any optimized backends (e.g. CPU, GPU, FPGA) at runtime.
+
+## Crate Organization
+
+### **poulpy-hal/layouts**
+
+This module defines backend agnostic layouts following **spqlios-arithmetic** types. There are two main types to distinguish from, user facing types and backend types, which are opaque write-only backend specific types. User facing types, such as `vec_znx`, serve both as the input and output of computations, while backend opaque types, such as `svp_ppol` (a.k.a. scalar vector product prepared polynomial), are backend specific representation for optimized evaluation. For example in a CPU implementation, an `svp_ppol`, which is the prepared type of `scalar_znx`, could be a `scalar_znx` stored in DFT with an AVX optimized data ordering.
+
+This module also provide various helpers over these types, as well as serialization for the front end types `scalar_znx`, `vec_znx` and `mat_znx`.
+
+#### Module
+
+The `module` is a struct that stores backend-specific pre-computations (e.g. DFT tables).
+
+#### ScalarZnx
+
+A `scalar_znx` is a front-end generic and backend agnostic type that stores a single small polynomial of `i64` coefficients. This type is mainly used to store secret-keys or small plaintext polynomials (for example GGSW plaintexts).
```

Comment:
> A `scalar_znx` is a front-end generic and backend agnostic type that stores a single small polynomial of `i64` coefficients.

Isn't `scalar_znx` struct generic over the data type? It may store small polynomial as i64 but it can also be some other representation ( for eg bytearray )?